### ISSUE TITLE
Unable to translate validation exception

### ISSUE THREAD

**Issue by FaridAghili** [MarcinOrlowski/laravel-api-response-builder#114]
**Describe the bug**
When Laravel validation fails and HTTP code 422 is returned, I'm getting something like this:
```json
    "success": false,
    "code": 115,
    "locale": "fa",
    "message": "The given data was invalid.",
    "data": ...
```

**Expected behavior**
How can I translate the `The given data was invalid.`?
I'm using v6.3.1 of this package

**Environemnt:**
 - Name and version of OS: Windows 10
 - PHP version 7.3
 - Laravel version 6.5.0

**Comment by MarcinOrlowski** [comment]
The `The given data was invalid` is default message for Laravel's ValidationException: (see constructor in 
`vendor/laravel/framework/src/Illuminate/ValidationException.php`). Where is this exception coming from? How it gets thrown?

**Comment by FaridAghili** [comment]
Exception is thrown through the FormRequest validation, but I thought we have control over exception message localization as the response `code` is `115' and I have already mapped that:

```php
'map' => [
    115 => 'api.VALIDATION_EXCEPTION',
],
```

Also what's the point of `exception_handler` config key then?

```php
'exception_handler' => [
    'exception' => [
        'validation_exception' => [
            'code' => ApiCodes::VALIDATION_EXCEPTION,
            'http_code' => Symfony\Component\HttpFoundation\Response::HTTP_UNPROCESSABLE_ENTITY,
        ],
    ],
],
```

**Comment by MarcinOrlowski** [comment]
In case of exceptions the `map` entry will only be used if exception comes with no message. If there's any message (so `$ex->getMessage()` returns non empty string), then that string takes precedence and will be returned as-is. In case there's no exception message provided, mapping will be used. You can try that yourself by throwing ValidationException with empty message:

    throw new ValidationException('');

It's works that's way because in most cases if exception comes with the message, then that message is specifically related to the cause of the problem that caused exception to be thrown which should be much more useful than always returning generic message, no matter what really happened.

**Comment by FaridAghili** [comment]
May I suggest to give priority to user defined message instead of `$ex->getMessage()`? Because some exception messages, including the one from `ValidationException` are hard coded in Laravel and cannot be localized, but your package can fix this limitation by honoring the mapped `code` translation.

**Comment by MarcinOrlowski** [comment]
Good idea. I will add and option for each exception that would allow to enforce specified over exception's one.

**Comment by FaridAghili** [comment]
Great, looking forward to it.
Thanks.

________________________________

**Comment by MarcinOrlowski** [comment]
I just merged this feature into `v6.4-dev` branch. It adds `msg_key` and `msg_enforce` keys to exception handler config. 

 -  `msg_key`    : (string) optional localization string key (ie. 'app.my_error_string') which will be used
          if exception's message is empty. If `msg_key` is not provided, ExceptionHandler will             fall back to built-in message.
  - `msg_enforce`: (boolean) if `true`, then fallback message (either one specified with `msg_key`, or
                built-in one will **always** be used, ignoring exception's message string completely.                 If set to `false` (default) then it will enforce either built-in message (if no                 `msg_key` is set, or message referenced by `msg_key` completely ignoring exception                message ($ex->getMessage()).

See config file for details: https://github.com/MarcinOrlowski/laravel-api-response-builder/blob/9683c61726586c5247741a319821cc375e240619/config/response_builder.php#L68
